### PR TITLE
More IResult Implementations

### PR DESCRIPTION
**Pull request by ilkayilknur**
Contributes https://github.com/dotnet/aspnetcore/issues/32565

`IResult` implementations for
- SignInResult
- SignOutResult
- ForbidResult
- ChallengeResult

### CODE REVIEW

File: src/Mvc/Mvc.Core/src/ChallengeResult.cs [hunk 1]
```diff
@@ -97,7 +98,18 @@ public override async Task ExecuteResultAsync(ActionContext context)
...
             }
 
-            var loggerFactory = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            await ExecuteAsync(context.HttpContext);
+        }
+
+        /// <inheritdoc />
+        async Task IResult.ExecuteAsync(HttpContext httpContext)
+        {
+            await ExecuteAsync(httpContext);
```
Comment (davidfowl) [inline src/Mvc/Mvc.Core/src/ChallengeResult.cs:107]: Remove the async/await, just pass through

File: src/Mvc/Mvc.Core/src/ForbidResult.cs [hunk 2]
```diff
@@ -97,7 +98,18 @@ public override async Task ExecuteResultAsync(ActionContext context)
...
             }
 
-            var loggerFactory = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            await ExecuteAsync(context.HttpContext);
+        }
+
+        /// <inheritdoc />
+        async Task IResult.ExecuteAsync(HttpContext httpContext)
+        {
+            await ExecuteAsync(httpContext);
```
Comment (davidfowl) [inline src/Mvc/Mvc.Core/src/ForbidResult.cs:107]: Same as above

File: src/Mvc/Mvc.Core/src/ChallengeResult.cs (hunk 1 above)
Comment (ilkayilknur) [reply src/Mvc/Mvc.Core/src/ChallengeResult.cs]: Done.

File: src/Mvc/Mvc.Core/src/ForbidResult.cs (hunk 2 above)
Comment (ilkayilknur) [reply src/Mvc/Mvc.Core/src/ForbidResult.cs]: Done.

**Review by davidfowl — APPROVED** [review]

File: src/Mvc/Mvc.Core/src/ChallengeResult.cs [hunk 3]
```diff
@@ -97,7 +98,18 @@ public override async Task ExecuteResultAsync(ActionContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var loggerFactory = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            await ExecuteAsync(context.HttpContext);
```
Comment (davidfowl) [inline src/Mvc/Mvc.Core/src/ChallengeResult.cs:101]: Also here, remove the extra state machine here.

**Comment by davidfowl** [comment]
One more nit

**Comment by ilkayilknur** [comment]
Thanks, @davidfowl. I've updated the related codes.

File: src/Mvc/Mvc.Core/src/ForbidResult.cs [hunk 4]
```diff
@@ -97,7 +98,18 @@ public override async Task ExecuteResultAsync(ActionContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var loggerFactory = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            await ExecuteAsync(context.HttpContext);
```
Comment (davidfowl) [inline src/Mvc/Mvc.Core/src/ForbidResult.cs:101]: Did you forget this one?

Comment (ilkayilknur) [reply src/Mvc/Mvc.Core/src/ForbidResult.cs]: yes, my bad. I removed the async/await

**Comment by davidfowl** [comment]
Sweet thanks!

**Pull request merged**